sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1739,6 +1739,16 @@ class DataSourceV2SQLSuiteV1Filter
     }
   }
 
+  test("SPARK-48709: varchar resolution mismatch for DataSourceV2 CTAS") {
+    withSQLConf(
+      SQLConf.STORE_ASSIGNMENT_POLICY.key -> SQLConf.StoreAssignmentPolicy.LEGACY.toString) {
```

Comment:
just for my education, why it's only a problem with the legacy store assignment?

Reply:
Maybe we can test all options here

Reply:
We will validate store assignment policy if `!v2Write.outputResolved` is true.
https://github.com/apache/spark/blob/0864bbeac399825619375187b6c0ddca3f408f90/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L3530-L3532

Reply:
How does this end up with the error `org.apache.spark.sql.AnalysisException: LEGACY store assignment policy is disallowed in Spark data source V2.`?

Reply:
I mean, it looks like the right error.

Reply:
oh nvm, when we don't need to do store assignment, then the mode doesn't matter.